### PR TITLE
#162949006 Sharing articles via Email and social media platforms

### PR DESCRIPTION
**Pull request by Nicanor008**
#### What does this PR do?
This PR merges the sharing articles feature branch to develop

#### Description of Task to be completed?
As an authenticated user, I can share an article directly to a specific user via email, or on social platforms that is twitter and facebook

#### How should this be manually tested?
Run the server and run the following endpoints in the postman
- POST ` http://127.0.0.1:8000/api/v1/test-articles/share/email/` to share an article with a specific user, enter their email on the body like 
```
{ "email":"sendingto@email.com"}
```
- POST ` http://127.0.0.1:8000/api/v1/<slug>/share/facebook/` to share an article via Facebook
- POST ` http://127.0.0.1:8000/api/v1/<slug>/share/twitter/` 

#### What are the relevant pivotal tracker stories?
[162949006 Users should be able to share articles across different channels](https://www.pivotaltracker.com/story/show/162949006)

##### Email share endpoint
screen shot 2019-01-31 at 12 38 40

##### Article shared in the email
screen shot 2019-01-31 at 12 28 05

##### Facebook article sharing endpoint
screen shot 2019-01-31 at 12 29 21

##### On sharing article on facebook
screen shot 2019-01-31 at 12 33 32

##### Twitter article sharing endpoint in postman
screen shot 2019-01-31 at 12 29 44

##### How to share on twitter and just tweet it
screen shot 2019-01-31 at 12 34 24

### CODE REVIEW

File: authors/apps/articles/serializers.py [hunk 1]
```diff
@@ -210,3 +210,12 @@ class ReportArticlesSerializer(serializers.ModelSerializer):
     class Meta:
         model = ReportArticle
         fields = ('article', 'user', 'report_msg')
+
+        
+class EmailCheckSerializer(serializers.Serializer):
+    email = serializers.EmailField(
+        required=True,
+        error_messages={
+            'invalid': 'Must be of email format'
```
Comment (charitymarani) [inline authors/apps/articles/serializers.py]: What's the email format? Make this more descriptive

Comment (Nicanor008) [reply authors/apps/articles/serializers.py]: on it

Comment (Nicanor008) [reply authors/apps/articles/serializers.py]: @charitymarani have implemented more descriptive message, thanks for the feedback

File: authors/apps/authentication/views.py [hunk 2]
```diff
@@ -62,10 +63,12 @@ def post(self, request):
...
             'username': username,
         })
-        body_content = strip_tags(message)
-        msg = EmailMultiAlternatives(subject, body_content, sender, to=[email])
-        msg.attach_alternative(message, "text/html")
-        msg.send()
+        # body_content = strip_tags(message)
+        # msg = EmailMultiAlternatives(subject, body_content, sender, to=[email])
+        # msg.attach_alternative(message, "text/html")
+        # msg.send()
```
Comment (kzyangiro) [inline authors/apps/authentication/views.py]: @Nicanor008 Unwanted comments here.

File: authors/apps/authentication/views.py [hunk 3]
```diff
@@ -62,11 +63,7 @@ def post(self, request):
             'title': subject,
             'username': username,
         })
-        body_content = strip_tags(message)
-        msg = EmailMultiAlternatives(subject, body_content, sender, to=[email])
-        msg.attach_alternative(message, "text/html")
-        msg.send()
-
+        base_mail_send(subject, sender, email, message)
```
Comment (eduhmik) [inline authors/apps/authentication/views.py]: Hey @Nicanor008 , good use of the base_mail_send() function to send all emails in our application. Keep up on **Good Code Refactor**

File: authors/apps/authentication/views.py (hunk 2 above)
Comment (Nicanor008) [reply authors/apps/authentication/views.py]: implemented @kzyGit

**Review by Nyakaru — APPROVED** [review]

**Review by JoshuaKodhe — APPROVED** [review]

**Pull request merged**